### PR TITLE
[Network] az network application-gateway waf-policy managed-rule rule-set: Expose server error to user to give more intuitive hint message

### PR DESCRIPTION
**Pull request by haroldrandom**
**Description**  

Fix https://github.com/Azure/azure-cli/issues/15019

**Testing Guide**  

**History Notes**  

[Network] az network application-gateway waf-policy managed-rule rule-set add: Expose server error to user to give more intuitive hint message.  
[Network] az network application-gateway waf-policy managed-rule rule-set update: Support to change rule set type version.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -1601,43 +1601,61 @@ def add_waf_managed_rule_set(cmd, client, resource_group_name, policy_name,
...
     """
     Update(Override) existing rule set of a WAF policy managed rules.
     """
-    ManagedRuleGroupOverride, ManagedRuleOverride = cmd.get_models('ManagedRuleGroupOverride', 'ManagedRuleOverride')
+    ManagedRuleSet, ManagedRuleGroupOverride, ManagedRuleOverride = \
+        cmd.get_models('ManagedRuleSet', 'ManagedRuleGroupOverride', 'ManagedRuleOverride')
 
     managed_rule_overrides = [ManagedRuleOverride(rule_id=r) for r in rules] if rules else None
 
     rule_group_override = ManagedRuleGroupOverride(rule_group_name=rule_group_name,
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:1631]: If `rule_group_name` can be `None`, How to deal with this line?

File: src/azure-cli/azure/cli/command_modules/network/_help.py [hunk 2]
```diff
@@ -1344,13 +1344,24 @@
 short-summary: >
   Add managed rule set to the WAF policy managed rules. For rule set and rules, please visit:
   https://docs.microsoft.com/en-us/azure/web-application-firewall/ag/application-gateway-crs-rulegroups-rules
+examples:
+  - name: Disable a attack protection rule
```
[the remark this answers is not in the document]
Comment (haroldrandom) [reply src/azure-cli/azure/cli/command_modules/network/_help.py:1348]:
```suggestion
  - name: Disable an attack protection rule
```

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 1 above)
Comment (haroldrandom) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:1631]: If `rule_group_name ` is None, `updated_rule_set` will be set to the target rule set in line if statement at line 1647 and for-loop get break at line 1648.

Then, the target rule set will be removed at line 1657 and a new MangedRuleSet will be appended to.

However, the `rule_group_overrides` property of new added ManagedRuleSet shoule contain an invalid ManagedRuleGroupOverride model if user don't provide --rules either.

If user prvode `--rules` without `--rule-group-name`, then the ManagedRuleGroupOverride model is invalid, server will reject this kind of request.

Comment (haroldrandom) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:1631]: Add validator for it

**Review by MyronFanQiu — APPROVED** [review]

**Pull request merged**